pyccel/ast/basic.py

```diff
@@ -308,49 +308,35 @@ class PyccelAstNode(Basic):
     """Class from which all nodes containing objects inherit
     """
     stage      = None
-    __slots__ = ('_shape', '_rank', '_dtype', '_precision', '_order')
-
-    def __init__(self):
-        super().__init__()
-        if not hasattr(self, '_shape'):
-            self._shape = None
-        if not hasattr(self, '_rank'):
-            self._rank = None
-        if not hasattr(self, '_dtype'):
-            self._dtype = None
-        if not hasattr(self, '_precision'):
-            self._precision = None
-        if not hasattr(self, '_order'):
-            self._order = None
 
```

Comment:
I think that you still need to set `__slots__ = ()` in this class, in order to avoid the creation of empty dictionaries `__dict__ = {}` in the instances of the subclasses.

Reply:
You are right. Fixed in 57df337